pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -1667,6 +1663,11 @@ func (og *operationGenerator) nodeExpandVolume(volumeToMount VolumeToMount, rsOp
 				// On retry, NodeExpandVolume will be called again but do nothing
 				return false, fmt.Errorf("MountVolume.NodeExpandVolume update PVC status failed : %v", err)
 			}
+			ok := actualStateOfWorld.MarkVolumeAsResized(volumeToMount.VolumeName, pvSpecCap)
```

Comment:
In the readonly case above, or any other case where we return true early, do we need to mark volume as resized? https://github.com/kubernetes/kubernetes/pull/92932/files#diff-450e811a4953f760ff1594ede8b2037eR1637

Reply:
we don't need to mark them as resized as such because such volumes are not even entered into DSOW's `persistentVolumeSize` field. 

for readonly volumes, we don't track their sizes in DSOW - https://github.com/kubernetes/kubernetes/pull/92932/files#diff-94e527d26427ceb4f990f05f76952656 and hence will not result in resize. I have also added an unit test that covers readonly volumes

Reply:
Would we still need to update the pvc status to say resizing is finished?

Reply:
we can't mark PVC resizing as finished because if PVC was readonly on this node, resizing will never happen whereas it can successfully happen on some other node.

Reply:
Having said that - all the `return true, blah` statements look weird but we are basically double checking scenarios we already checking when a volume enters DSOW for volume expansion. May be that can be cleaned up but it does not hurt to be safe.